### PR TITLE
restaurant/feature/add_cuisine

### PR DESCRIPTION
**Pull request by aya-seif**

### CODE REVIEW

**Review by devsadeq — APPROVED** [review]
LGTM

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/CuisineRoutes.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.thechance.service_restaurant.usecase.cuisine.CuisineUseCasesContainer
+
+fun Route.cuisineRoutes() {
+
+    val cuisineUseCasesContainer: CuisineUseCasesContainer by inject()
+
+    route("cuisine") {
+
+        post {
+            val cuisine = call.receive<CuisineCollection>()
```
Comment (Asiasama710) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/CuisineRoutes.kt:26]: CuisineDto not CuisineCollection

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/CuisineRoutes.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            val isAdded = cuisineUseCasesContainer.addCuisineUseCase(cuisine.toEntity())
```
Comment (Asiasama710) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/CuisineRoutes.kt:27]: i think` toEntity() `not here but in usecase

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/CuisineRoutes.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        get("/{id}") {
+            val id = call.parameters.extractString("id") ?: throw NotFoundException()
+            val cuisine = cuisineUseCasesContainer.getCuisineByIdUseCase(id)
+            call.respond(cuisine.toDto())
+        }
+
+        put {
+            val cuisine = call.receive<CuisineDto>()
+            val isUpdated = cuisineUseCasesContainer.updateCuisineUseCase(cuisine.toEntity())
```
Comment (Asiasama710) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/CuisineRoutes.kt:46]: i think `toEntity()` not here but in usecase

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/MealsRoutes.kt [hunk 4]
```diff
@@ -37,6 +28,13 @@ fun Route.mealRoutes() {
...
+        get {
+            val page = call.parameters.extractInt("page") ?: 1
+            val limit = call.parameters.extractInt("limit") ?: 10
+            val meals = mealUseCasesContainer.getMealsUseCase(page, limit)
+            call.respond(meals.toDto())
+        }
+
         post {
             val meal = call.receive<MealCollection>()
             val isAdded = mealUseCasesContainer.addMealUseCase(meal.toEntity())
```
Comment (Asiasama710) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/MealsRoutes.kt:40]: i think `toEntity()` not here but in usecase

File: service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/MealsRoutes.kt [hunk 5]
```diff
@@ -37,6 +28,13 @@ fun Route.mealRoutes() {
...
 
+        get {
+            val page = call.parameters.extractInt("page") ?: 1
+            val limit = call.parameters.extractInt("limit") ?: 10
+            val meals = mealUseCasesContainer.getMealsUseCase(page, limit)
+            call.respond(meals.toDto())
+        }
+
         post {
             val meal = call.receive<MealCollection>()
```
Comment (Asiasama710) [inline service_restaurant/src/main/kotlin/org/thechance/service_restaurant/api/endpoints/MealsRoutes.kt:39]: `MealCollection `should be MealDto

**Review by Asiasama710 — CHANGES_REQUESTED** [review]

**Pull request merged**